server.go

```diff
@@ -404,3 +404,49 @@ func scannerDeregister(joinIP string, joinPort uint16, id string) error {
 	}
 	return nil
 }
+
+func getScannerAvailable(joinIP string, joinPort uint16, data *share.ScannerRegisterData, cb cluster.GRPCCallback) (*share.ScannerAvailable, error) {
+	client, err := getControllerServiceClient(joinIP, joinPort, cb)
+	if err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("Failed to find ctrl client")
+		return &share.ScannerAvailable{Visible: false}, errors.New("Failed to connect to controller")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
+	defer cancel()
+
+	scannerAvailable, errHealthCheck := client.HealthCheck(ctx, data)
+
+	return scannerAvailable, errHealthCheck
+}
+
+// To ensure the controller's availability, periodCheckHealth use HealthCheck to periodically check if the controller is alive.
+// Additionally, if the controller is deleted or not responsive, the scanner will re-register.
+func periodCheckHealth(joinIP string, joinPort uint16, data *share.ScannerRegisterData, cb *clientCallback, healthCheckCh chan struct{}, done chan bool, period, retryMax int) {
```

Comment:
Optional: You can use context, which is the standard way to control go routine, to replace `healthCheckCh`. You can also assign timeout to it.

Reply:
I think I will remain the same, since we may trigger the re-register when we have the idle state.
Thus I think we should have a way to close the channel manually.